plugins/sim/cdsimcontroller.cpp

```diff
@@ -402,28 +491,45 @@ void CDSimController::ensureSimContactsPresent()
 
 void CDSimController::voicemailConfigurationChanged()
 {
-    if (!m_voicemailConf || !m_simManager.present()) {
+    ModemData *modem = 0;
+    if (MGConfItem *voicemailConf = qobject_cast<MGConfItem *>(sender())) {
```

Comment:
How about making `ModemData` (or its equivalent) a `Q_OBJECT` so that it can receive these signals directly? That way we wouldn't have to go through all these casts and lookups just to get to the associated `ModemData`.